src/test/test_pidfile.sh

```diff
@@ -26,8 +26,8 @@ function TEST_without_pidfile() {
     local dir=$1
     setup $dir
     local RUNID=`uuidgen`
-    run_mon $dir a --pid-file= --daemonize=$RUNID || { teardown_unexist_pidfile $dir; return 1; } 
-    run_osd $dir 0 --pid-file= --daemonize=$RUNID || { teardown_unexist_pidfile $dir; return 1; }
+    run_mon $dir a --pid-file= --daemonize=$RUNID || { teardown_unexist_pidfile $dir $RUNID; return 1; }
+    run_osd $dir 0 --pid-file= --daemonize=$RUNID || { teardown_unexist_pidfile $dir $RUNID; return 1; }
```

Comment:
@dachary  
 --daemonize=$RUNID passes to daemon by enviroment CEPH_ARGS, so we cant find out the osd daemon we boot by 'awk $RUNID', which leaves a osd daemon runnig after unittest finish.
suggestion: change the procesure run osd, then kill osd, finally run osd with uuid, but without pidfile, like this ./ceph-osd ... --pidfile= --daemonize=$RUNID..